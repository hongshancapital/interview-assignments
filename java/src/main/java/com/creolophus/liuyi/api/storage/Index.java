package com.creolophus.liuyi.api.storage;

import com.creolophus.liuyi.api.util.GoogleHash;
import java.util.Objects;

/**
 * @author magicnana
 * @date 2021/7/14 18:43
 */
public final class Index {


  private String shortUrl;
  private int position;
  private int length;
  private int index;
  private String longUrl;

  public Index(String shortUrl,int position,int length){
    int hash = Math.abs(GoogleHash.murmur3_32_int(shortUrl));
    this.index = hash < Config.INDEX_FILE_SIZE?hash:hash % Config.INDEX_FILE_SIZE;
    this.shortUrl = shortUrl;
    this.position = position;
    this.length = length;
  }

  public String getShortUrl() {
    return shortUrl;
  }

  public String getLongUrl() {
    return longUrl;
  }

  public void setLongUrl(String longUrl) {
    this.longUrl = longUrl;
  }

  public int getPosition() {
    return position;
  }

  public int getIndex() {
    return index;
  }

  public int getLength() {
    return length;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Index index1 = (Index) o;
    return getPosition() == index1.getPosition() &&
        getLength() == index1.getLength() &&
        getIndex() == index1.getIndex() &&
        Objects.equals(getShortUrl(), index1.getShortUrl());
  }

}
