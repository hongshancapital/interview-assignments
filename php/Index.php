<?php
namespace app\controller;

use app\BaseController;
use app\model\UserModel;
class Index extends BaseController
{
//    public function index()
//    {
//        return '<style type="text/css">*{ padding: 0; margin: 0; } div{ padding: 4px 48px;} a{color:#2E5CD5;cursor: pointer;text-decoration: none} a:hover{text-decoration:underline; } body{ background: #fff; font-family: "Century Gothic","Microsoft yahei"; color: #333;font-size:18px;} h1{ font-size: 100px; font-weight: normal; margin-bottom: 12px; } p{ line-height: 1.6em; font-size: 42px }</style><div style="padding: 24px 48px;"> <h1>:) </h1><p> ThinkPHP V' . \think\facade\App::version() . '<br/><span style="font-size:30px;">14载初心不改 - 你值得信赖的PHP框架</span></p><span style="font-size:25px;">[ V6.0 版本由 <a href="https://www.yisu.com/" target="yisu">亿速云</a> 独家赞助发布 ]</span></div><script type="text/javascript" src="https://tajs.qq.com/stats?sId=64890268" charset="UTF-8"></script><script type="text/javascript" src="https://e.topthink.com/Public/static/client.js"></script><think id="ee9b1aa918103c4fc"></think>';
//    }

    //登录页面
    public function login()
    {
       return view('/index');
    }

    /**注册方法
     * @param username string  账号
     * @param  passwd string  密码 //双md5加密方式
     * @return json
     */
    public function register()
    {
        //1.验证
        $check = $this->check();
        if($check['code']==400){
            return json_encode($check);
        }
        //2.数据库操作
        $data['username'] = $_POST['username'];
        $data['passwd'] = md5(md5($_POST['passwd']));
        $model = new UserModel();
        $res = $model->regster($data);
        if($res['code']== 200){
            return json_encode($res);
        }
        return json_encode($res);
    }

    /** 注册验证方法
     * @param username string  账号
     * @param  passwd string  密码
     * @return array
     */
    public function check(){
        if(empty($_POST['username'])){
            $json_data = [
                'code' => 400,
                'msg'  => '用户名不能为空',
                'data' => []
            ];
            return $json_data;
        }

        if(!preg_match("/^[a-zA-Z_](.*)?$/",$_POST['username']) ){
            $json_data = [
                'code' => 400,
                'msg'  => '用户名只能以英文字母或下划线开头',
                'data' => []
            ];
            return $json_data;
        }

        if(!preg_match("/^[a-zA-Z0-9_]+$/",$_POST["username"])){
            $json_data = [
                'code' => 400,
                'msg'  => '用户名只能包含英文字母，下划线或数字',
                'data' => []
            ];
            return $json_data;
        }

        if(empty($_POST['passwd'])){
            $json_data = [
                'code' => 400,
                'msg'  => '密码不能为空',
                'data' => []
            ];
            return $json_data;
        }

        $reg1 = '/([.*?]){6,}$/'; //密码必须是6位以上
        if(strlen($_POST['passwd'])<6){
            $json_data = [
                'code' => 400,
                'msg'  => '密码必须是6位以上',
                'data' => []
            ];
            return $json_data;
        }

        $reg2 = '/(123|234|345|456|567|678|789|012)/'; //不能有3个连续数字
        if(preg_match($reg2,$_POST['passwd'])){
            $json_data = [
                'code' => 400,
                'msg'  => '密码不能有3个连续数字',
                'data' => []
            ];
            return $json_data;
        }

        $reg3 ='/(?=.*\d)(?=.*[a-zA-Z])(?=.*[~!@#$%^&*.])/';//必须含有字母、数字、特殊符号
        if(!preg_match($reg3,$_POST['passwd'])){
            $json_data = [
                'code' => 400,
                'msg'  => '密码必须含有字母、数字、特殊符号',
                'data' => []
            ];
            return $json_data;
        }
        return ['code'=>200];
    }
}
