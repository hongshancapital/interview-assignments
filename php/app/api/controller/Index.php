<?php
declare (strict_types=1);

/**
 * 开发: Atom
 * 时间: 2021-05-15
 */

namespace app\api\controller;


use app\BaseController;

/**
 * 登陆
 * Class Index
 * @package app\api\controller
 */
class Index extends BaseController
{
    public function index(){

        return view('index');
    }
}