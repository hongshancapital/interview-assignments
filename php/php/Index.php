<?php
defined('BASEPATH') OR exit('No direct script access allowed');

class Index extends CI_Controller {

	public function __construct()
	{
		parent::__construct();
	}

	/**
	 * 注册页
	 */
	public function index()
	{
		$this->load->view('register');
	}
}
