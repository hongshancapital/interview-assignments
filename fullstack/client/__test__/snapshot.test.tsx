import { render } from '@testing-library/react'
import Index from '../src/pages/index'

it('renders homepage unchanged', () => {
  const { container } = render(<Index />)
  expect(container).toMatchSnapshot()
})