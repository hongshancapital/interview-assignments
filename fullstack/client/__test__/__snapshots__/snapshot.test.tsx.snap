// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <p>
    短域名服务
  </p>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-spinning css-dev-only-do-not-override-ixblex"
  >
    <span
      class="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
    </span>
  </div>
  <p>
    短域名读取
  </p>
  <span
    class="ant-input-group-wrapper ant-input-search ant-input-search-large ant-input-search-with-button ant-input-group-wrapper-lg css-dev-only-do-not-override-ixblex"
  >
    <span
      class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-ixblex"
    >
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-ixblex"
      >
        <input
          class="ant-input ant-input-lg css-dev-only-do-not-override-ixblex"
          placeholder="input search text"
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <span
            class="ant-input-clear-icon ant-input-clear-icon-hidden"
            role="button"
            tabindex="-1"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-primary ant-btn-lg ant-input-search-button"
          type="button"
        >
          <span>
            Search
          </span>
        </button>
      </span>
    </span>
  </span>
  <div />
  <p>
    短域名创建
  </p>
  <span
    class="ant-input-group-wrapper ant-input-search ant-input-search-large ant-input-search-with-button ant-input-group-wrapper-lg css-dev-only-do-not-override-ixblex"
  >
    <span
      class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-ixblex"
    >
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-ixblex"
      >
        <input
          class="ant-input ant-input-lg css-dev-only-do-not-override-ixblex"
          placeholder="input origin url"
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <span
            class="ant-input-clear-icon ant-input-clear-icon-hidden"
            role="button"
            tabindex="-1"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-ixblex ant-btn-primary ant-btn-lg ant-input-search-button"
          type="button"
        >
          <span>
            Create
          </span>
        </button>
      </span>
    </span>
  </span>
  <div />
</div>
`;
